src/messages/layouts/MessageLayout.cpp

```diff
@@ -139,11 +140,15 @@ void MessageLayout::actuallyLayout(int width, MessageElementFlags flags)
             continue;
         }
 
-        if (getSettings()->hideModerationActions &&
-            (this->message_->flags.has(MessageFlag::Timeout) ||
-             this->message_->flags.has(MessageFlag::Untimeout)))
+        if (this->message_->flags.has(MessageFlag::Timeout) ||
+            this->message_->flags.has(MessageFlag::Untimeout))
         {
-            continue;
+            if (getSettings()->hideModerationActions ||
+                (getSettings()->streamerModeSuppressModActions &&
+                 isInStreamerMode()))
+            {
+                continue;
+            }
```

Comment:
is the change of prioritising either check for relevant settings or check for relevant message flags intended? either way I guess it's okay since checking flags seems less expensive than calling settings instance

Reply:
this order seems preferable to potentially avoid an expensive `isInStreamerMode()` lookup (on a cache miss)

Reply:
Yeah, I made this order to be somewhat readable and avoid calling isInStreamerMode if not needed. I'll add a comment explaining why this is.